[wip] roachtest: repro branch for #68151

tbg opened the pull request:
See #68151.

tbg committed:
wip

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/68160)
<!-- Reviewable:end -->

tbg committed:
tpcc: fix error handling in check 3.3.2.4

A classic footgun baked into the `database/sql` is that there's an
easy-to-miss error check:

```go
rows, err := db.Query(/* ... */)
if err != nil { return err }
for rows.Next() { /* stuff */ }
if rows.Err() != nil { // <------------- footgun
  return err
}
return rows.Close()
```

This was the case in our impl of TPCC check 3.3.2.4. As of a recent
change, the second query used in that check runs into the SQL memory
budget in our nightly tests. The missing error check was making that
look like a scary anomaly.

The other checks all have the same problem. I'll file an issue
for that and link it to this PR.

Fixes #68151.

Release note: None
